compass/landice/tests/ismip6_forcing/ocean_basal/process_basal_melt.py

```diff
@@ -0,0 +1,230 @@
+        input_file_list = self._files_coeff
+        i = 0
+        for file in input_file_list:
+            print(f"processing the input file {os.path.basename(file)}")
+            i += 1
+            self.combine_ismip6_inputfiles(os.path.basename(self._file_basin),
+                                           os.path.basename(file),
+                                           combined_file_temp)
+
+            # remap the input forcing file.
+            print("Calling the remapping function...")
+            self.remap_ismip6BasalMelt_to_mali(combined_file_temp,
+                                               remapped_file_temp,
+                                               mali_mesh_name,
+                                               mali_mesh_file, method_remap)
+
+            output_file = f"processed_basin_and_{os.path.basename(file)}"
+
+            # rename the ismip6 variables to MALI variables
+            print("Renaming the ismip6 variables to mali variable names...")
+            self.rename_ismip6BasalMelt_to_mali_vars(remapped_file_temp,
+                                                     output_file)
+
+            print("Remapping and renamping process done successfully. "
+                  "Removing the temporary files 'combined.nc' "
+                  "and 'remapped.nc'")
+
+            # remove the temporary combined file
+            os.remove(combined_file_temp)
+            os.remove(remapped_file_temp)
+
+            # place the output file in appropriate directory
+            if output_base_path == "NotAvailable":
+                return
+
+            output_path = f'{output_base_path}/basal_melt/parametrizations/'
+            if not os.path.exists(output_path):
+                print("Creating a new directory for the output data")
+                os.makedirs(output_path)
+
+            src = os.path.join(os.getcwd(), output_file)
+            dst = os.path.join(output_path, output_file)
+            shutil.copy(src, dst)
+
+            print("")
+
+    def combine_ismip6_inputfiles(self, basin_file, coeff_gamma0_deltaT_file,
+                                  combined_file_temp):
+        """
+        Combine ismip6 input files before regridding onto mali mesh
+
+        Parameters
+        ----------
+        basin_file : str
+            imbie2 basin numbers in ismip6 grid
+        coeff_gamma0_deltaT_file : str
+            uniform melt rate coefficient (gamma0) and temperature
+            correction per basin
+        combined_file_temp : str
+            temporary output file that has all the variables combined
+        """
+
+        ds_basin = xr.open_dataset(basin_file, engine="netcdf4")
+        ds = xr.open_dataset(coeff_gamma0_deltaT_file, engine="netcdf4")
+
+        ds["ismip6shelfMelt_basin"] = ds_basin.basinNumber
+        write_netcdf(ds, combined_file_temp)
+
+    def remap_ismip6BasalMelt_to_mali(self, input_file, output_file,
```

Comment:
I'd really strongly prefer we avoid mixed case in method names.
 ```suggestion
     def remap_ismip6_basal_melt_to_mali(self, input_file, output_file,
```